Source/ImPlotWidgets/Public/ImPlotWidget.h

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) Coconut Lizard Limited. All rights reserved.
+#pragma once
+
+#include <Runtime/Core/Public/CoreMinimal.h>
+#include <ImGui/Public/Widgets/ImGuiCustomWidget.h>
+
+struct FDataPoint
+{
+	float X = 0.0f;
+	float Y = 0.0f;
+};
+
+class IMPLOTWIDGETS_API SImPlotWidget : public SCompoundWidget
+{
+public:
+	SLATE_BEGIN_ARGS(SImPlotWidget)
+	{}
+	SLATE_ARGUMENT(TArray<TArray<FDataPoint>>, DataArrays)
+	SLATE_ARGUMENT(TArray<float>, InData)
+	SLATE_ARGUMENT(TArray<char*>, DataSetNames)
+	SLATE_ARGUMENT(FString, GraphTitle)
+	SLATE_ARGUMENT(FString, XLabelName)
+	SLATE_ARGUMENT(FString, YLabelName)
+	// The ImGui Custom Context used by this Widget (Zero Indexed)
+	SLATE_ARGUMENT(int32, ContextIndex)
+	SLATE_END_ARGS()
+
+	virtual void Construct(const FArguments& InArgs);
+
+	int32 NextContextIndex();
+
+protected:
+
+	virtual void OnImGuiDraw() const;
+
+
+	TSharedPtr<SImGuiCustomWidget> ImGuiCustomWidget;
+
+	TArray<TArray<FDataPoint>> DataArrays;
+	TArray<float> InData;
+	TArray<TArray<float>> XValueArrays;
+	TArray<TArray<float>> YValueArrays;
+
+	TArray<char*> DataSetNames;
+
+	FString Title;
+	FString XLabel;
+	FString YLabel;
+
+	float XMaxValue;
+	float YMaxValue;
+	float YMinValue;
+
+	mutable TArray<bool> ShowLine;
+
+	bool* CloseButtonVisible = nullptr;
+
+private:
```

Comment:
Probably worth just tidying this up a bit and removing this dangling private: and the whitespaces around it